workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,54 @@
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  interface IteratorConstructor {
+    from<T>(
+      object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+    ): IterableIterator<T>;
+  }
+
+  const Iterator: IteratorConstructor;
+}
+
+(globalThis as Record<string, unknown>).Iterator ??= {};
+
+(globalThis as Record<string, unknown>).Iterator = {
```

Comment:
This is not quite safe because it will _replace_ any existing `Iterator` object with the below. Similar to the other built-ins we're trying to fill in, we should do it conditionally.

So something like `globalThis.Iterator.from ??= function<T>(object: ...) { ... }` with the necessary casts to make that work.

Reply:
when i was doing the `??=` i was getting this error for all the test cases below

![6d005fdf176bd77fc28c45d67f490703](https://github.com/user-attachments/assets/5c64d9a9-3956-4170-b9d8-9a7ccd302b41)

Reply:
Can you try `console.log`-ing what `Iterator` is and what `Iterator.from` is in those scenarios? We do need to figure out a way to use `??=`.

Reply:
(Note that the other comments aren't blocked on this, so if this is the only thing left we can try to move forward as-is for now.)

Reply:
I was able to get things to work locally with the following:

```ts
(globalThis as Record<string, unknown>).Iterator ??= {};

((globalThis as Record<string, unknown>).Iterator as { from: unknown }).from ??=
  function <T>(
...
```

Reply:
doing this leads me to 

![bdcc30361c65f711c2ee5501ebd9ba0a](https://github.com/user-attachments/assets/ad05162c-3663-423b-9e12-d34cc3facead)

Reply:
actually got it working!